mne/epochs.py

```diff
@@ -2640,6 +2696,134 @@ def _get_epoch_from_raw(self, idx, verbose=None):
+                                   ' the raw object.')
+            new_annotations = annotations.copy()
+            new_annotations._prune_ch_names(self.info, on_missing)
+            self._annotations = new_annotations
+
+        return self
+
+    def get_epoch_annotations(self):
+        """Return a list of raw annotations per epoch.
+
+        Returns
+        -------
+        epoch_annots : list
+            A list (with length equal to number of epochs) of annotations per
+            epoch. Each annotation is stored as (onset, duration, description),
+            where the onset is now relative to the epoch, rather then in
+            continuous time.
+        """
+        # check if annotations exist
+        if self.raw_annotations is None:
+            return self
+
+        events = self.events
+
+        # create a list of annotations for each epoch
+        epoch_annot_list = [[] for _ in range(len(events))]
+
+        for annot in self.raw_annotations:
+            onset_ = annot['onset']
+            duration_ = annot['duration']
+            description_ = annot['description']
+
+            # convert onset to samples and account for first time
+            onset_samp = int(onset_ * self._raw_sfreq + self._first_time)
+            duration_samp = int(duration_ * self._raw_sfreq)
+
+            # loop through events to see which Epochs this annotation
+            # belongs to based on the onset and duration
+            epoch_index = _get_epoch_index_of_annot(
+                self, onset_samp, duration_samp)
+
+            for eidx in epoch_index:
+                # now convert onset sample relative to the Epoch and convert
+                # back to seconds relative to the Epoch
+                onset_samp_ = onset_samp - events[eidx, 0]
+                onset_ = onset_samp_ / self._raw_sfreq
+
+                epoch_annot_list[eidx].append(
+                    (onset_, duration_, description_))
+        return epoch_annot_list
+
+    def add_annotations_to_metadata(self):
+        """Add raw annotations into the Epochs metadata data frame.
+
+        Returns
+        -------
+        self : instance of Epochs
+            The modified instance (instance is also modified inplace).
+        """
+        pd = _check_pandas_installed()
+
+        # check if annotations exist
+        if self.annotations is None:
+            return self
+
+        # create a list of annotations for each epoch
+        epoch_annot_list = [[] for _ in range(len(self))]
+
+        # loop through each annotations and add to the metadata
+        if self._metadata is not None:
+            metadata = self._metadata
+        else:
+            data = np.empty((len(self), 0))
+            metadata = pd.DataFrame(data=data)
+
+        # get the Epoch annotations
+        epoch_annot_list = self.get_epoch_annotations()
+
+        # create a new Annotations column
+        metadata['Annotations'] = [[] for _ in range(metadata.shape[0])]
```

Comment:
why not directly assign `metadata['Annotations'] = epoch_annot_list` here?  or convert `epoch_annot_list` to a pandas Series and then add the whole thing as a column afterwards?

Reply:
I don't think it works syntatcitcally, but I'll try a few different iterations.

Reply:
Okay pd.Series works.